### DISCUSSION TITLE
Simple Commerce with Livewire

### DISCUSSION THREAD

**Discussion by sstottelaar** [duncanmcclean/simple-commerce#1158] [Help] [answered]
We have a project in the pipeline that we plan to build using Statamic and Livewire, mainly for interactivity and a better UX. I’ve been prototyping a few ideas (like adding products to the cart), and everything seems to be working fine so far.

However, I’m starting to question whether this approach makes sense in the long run. It feels like I’m rewriting a lot of logic that’s already available in the package (e.g., checking if an item is already in the cart, etc.).

Has anyone here built a complete shop with Simple Commerce and Livewire integration? If so, I’d love to hear your thoughts or tips on how to approach this. Any feedback would be much appreciated!

Cheers!

**Comment by duncanmcclean** [comment, marked as answer]
I never really intended for people to do "things" themselves, like in Livewire components, rather than use the `<form>`s generated by Simple Commerce.

You can do all of that stuff yourself, but like you say, you need to copy a lot of code into your own project. I'll see if there's a way I can simplify that in the [next version of Simple Commerce](https://duncanmcclean.com/future-of-simple-commerce).
